### PR TITLE
[New Rule] GCP Kubernetes Rolebindings Created or Patched

### LINKED ISSUE #1191: [New Rule] Kubernetes Creation/Patching of RoleBindings/ClusterRoleBindings
## Description

Kubernetes is an attacking vector that current there is no detection rules for. During my initial analysis I was able to create a KQL queries that can detect the following activity:

The purpose of this rule is to detect any new rolebindings or clusterrolebindings or updating the current ones within the cluster. The main idea behind this rule is to detect the creation of a malicious rolebinding that could be created by compromised account/service account for elevation of privileges.

### Target indexes

`filebeat-*` with `gcp` module having `audit` configured.

### Target Operating Systems

Kubernetes

### Platforms

Google Kubernetes Engine

### Tested ECS Version
1.8.0

### Query

In this query the only service account that would be ignored is the `system:addon-manager`

<details>
<summary>Query</summary>

```
"query": {
    "bool": {
      "must": [],
      "filter": [
        {
          "bool": {
            "should": [
              {
                "bool": {
                  "should": [
                    {
                      "match_phrase": {
                        "gcp.audit.method_name": "io.k8s.authorization.rbac.v1.clusterrolebindings.create"
                      }
                    }
                  ],
                  "minimum_should_match": 1
                }
              },
              {
                "bool": {
                  "should": [
                    {
                      "match_phrase": {
                        "gcp.audit.method_name": "io.k8s.authorization.rbac.v1.rolebindings.create"
                      }
                    }
                  ],
                  "minimum_should_match": 1
                }
              },
              {
                "bool": {
                  "should": [
                    {
                      "match_phrase": {
                        "gcp.audit.method_name": "io.k8s.authorization.rbac.v1.clusterrolebindings.patch"
                      }
                    }
                  ],
                  "minimum_should_match": 1
                }
              },
              {
                "bool": {
                  "should": [
                    {
                      "match_phrase": {
                        "gcp.audit.method_name": "io.k8s.authorization.rbac.v1.rolebindings.patch"
                      }
                    }
                  ],
                  "minimum_should_match": 1
                }
              }
            ],
            "minimum_should_match": 1
          }
        },
        {
          "range": {
            "@timestamp": {
              "gte": "2021-05-04T13:44:23.264Z",
              "lte": "2021-05-05T13:44:23.264Z",
              "format": "strict_date_optional_time"
            }
          }
        },
        {
          "match_phrase": {
            "event.module": "gcp"
          }
        },
        {
          "match_phrase": {
            "gcp.audit.type": "type.googleapis.com/google.cloud.audit.AuditLog"
          }
        },
        {
          "match_phrase": {
            "log.logger": "projects/elastic-infosec/logs/cloudaudit.googleapis.com%2Factivity"
          }
        }
      ],
      "should": [],
      "must_not": [
        {
          "match_phrase": {
            "gcp.audit.authentication_info.principal_email": "system:addon-manager"
          }
        }
      ]
    }
  }
```

</details>

### Related issues or PRs

Ready to create the necessary detection rule `toml` files

### References

https://cloud.google.com/kubernetes-engine/docs/how-to/audit-logging

### PR DESCRIPTION
**Pull request by austinsonger**
## Issues

Resolves #1191

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/integrations/gcp/credential_access_gcp_kubernetes_rolebindings_creation.toml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+[metadata]
+creation_date = "2021/06/06"
+maturity = "production"
+updated_date = "2021/06/06"
+
```
Comment (w0rk3r) [inline rules/integrations/gcp/credential_access_gcp_kubernetes_rolebindings_creation.toml:5]: Missing Integration

File: .gitignore [hunk 2]
```diff
@@ -8,6 +8,7 @@
 /*.yml
 /config
 /junit.xml
+/elastic-detection-cli
```
Comment (w0rk3r) [inline .gitignore:11]:
```suggestion
```

File: rules/integrations/gcp/credential_access_gcp_kubernetes_rolebindings_creation.toml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+[rule]
+author = ["Elastic","Austin Songer"]
```
Comment (w0rk3r) [inline rules/integrations/gcp/credential_access_gcp_kubernetes_rolebindings_creation.toml:7]:
```suggestion
author = ["Elastic", "Austin Songer"]
```

File: rules/integrations/gcp/credential_access_gcp_kubernetes_rolebindings_creation.toml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+description = """
+Identifies the creation or patching of potential malicious rolebinding.
```
Comment (w0rk3r) [inline rules/integrations/gcp/credential_access_gcp_kubernetes_rolebindings_creation.toml:9]: How we differentiate from normal activity? If not, needs to be reformulated, also needs some context around rolebindings

File: rules/integrations/gcp/credential_access_gcp_kubernetes_rolebindings_creation.toml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+"""
+from = "now-20m"
+index = ["filebeat-*", "logs-gcp*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "GCP "
```
Comment (w0rk3r) [inline rules/integrations/gcp/credential_access_gcp_kubernetes_rolebindings_creation.toml:15]: Missing Rule name

File: rules/integrations/gcp/credential_access_gcp_kubernetes_rolebindings_creation.toml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+note = """## Config
+
+The GCP Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule."""
+references = [
+    "https://cloud.google.com/kubernetes-engine/docs/how-to/audit-logging",
```
Comment (w0rk3r) [inline rules/integrations/gcp/credential_access_gcp_kubernetes_rolebindings_creation.toml:20]: Needs a more specific/related reference

File: rules/integrations/gcp/credential_access_gcp_kubernetes_rolebindings_creation.toml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+
+[rule.threat.tactic]
+id = "TA0006"
+name = "Credential Access"
+reference = "https://attack.mitre.org/tactics/TA0006/"
+
```
Comment (w0rk3r) [inline rules/integrations/gcp/credential_access_gcp_kubernetes_rolebindings_creation.toml:43]: Can't see a case for Credential Access here

**Review by w0rk3r — CHANGES_REQUESTED** [review]
Left some points we need to clarify, thanks!

File: rules/integrations/gcp/privilege_escalation_gcp_kubernetes_rolebindings_created_or_patched.toml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+]
+risk_score = 47
+rule_id = "2f0bae2d-bf20-4465-be86-1311addebaa3"
+severity = "medium"
+tags = ["Elastic", "Cloud", "GCP", "Continuous Monitoring", "SecOps", "Configuration Audit"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.dataset:(googlecloud.audit or gcp.audit) and event.action: (io.k8s.authorization.rbac.v*.clusterrolebindings.create or
```
Comment (brokensound77) [inline rules/integrations/gcp/privilege_escalation_gcp_kubernetes_rolebindings_created_or_patched.toml:34]:
```suggestion
event.dataset:(googlecloud.audit or gcp.audit) and event.action:(io.k8s.authorization.rbac.v*.clusterrolebindings.create or 
```

**Comment by CyberTaoFlow** [comment]
@austinsonger Can we have the system:addon-manager user provided as an exception by default for this otherwise its quite noisy.

**Pull request merged**